### ISSUE TITLE
Has GC_INTERVAL gone missing?

### ISSUE THREAD

**Issue by alexmbird** [thumbor/thumbor#1181]
### Thumbor request URL
n/a

### Expected behaviour
A config parameter `GC_INTERVAL` is listed in Thumbor's documentation [here](https://thumbor.readthedocs.io/en/latest/configuration.html#gc-interval).  It was added from [this PR](https://github.com/thumbor/thumbor/pull/905) back in 2017.  I've been attempting to use this while troubleshooting high memory usage on our Thumbor pods.

### Actual behaviour
But the code in `thumbor/server.py` to implement timed garbage collection has since disappeared, and grepping the codebase for any use of `GC_INTERVAL` returns no matches.  Is `GC_INTERVAL` still supported?  

### Operating system
n/a

### Your thumbor.conf
n/a

**Comment by kkopachev** [comment]
Is the memory leak still an issue?
There is being some work to break circular references which helps with garbage collection, so there should be less garbage hold back after each request. See https://github.com/thumbor/thumbor/pull/1003

**Comment by alexmbird** [comment]
Here's the memory usage I'm seeing on our platform.  We use Thumbor to resize images on the fly then serve them to our CDN (which mostly caches them) and we serve about 2 requests per second.  As you can see memory use stabilises after 12h or so, but given the steady trickle of requests I wouldn't expect usage to go on increasing for so long after startup.

![thumbor_memory](https://user-images.githubusercontent.com/691831/54449787-d34f8e80-4746-11e9-96af-ba2aec486e03.gif)

I'm running the latest Thumbor release, `6.7.0`.

**Comment by kkopachev** [comment]
What kind of loader/result storage do you use?

**Comment by alexmbird** [comment]
* Loader - `thumbor.storages.no_storage`, we pull original images from a bucket nearby via HTTP
* Result - `thumbor.result_storages.file_storage` on a local volume with caching for 60 seconds

**Comment by rafaeleyng** [comment]
There's really no use of the `GC_INTERVAL` configuration anywhere in the code. Besides the memory leak that you've reported, I think we should remove that from docs and from [config.py](https://github.com/thumbor/thumbor/blob/f4bf80ae156189db41f834dd7760061d88bcf713/thumbor/config.py#L149). I can remove those if you are ok with that @kkopachev.

**Comment by kkopachev** [comment]
I am not sure we should remove it. Looks like it is useful in some cases. Probably I've being way too aggressive with removing the code around it?

Since memory is not constantly growing and has an upper threshold (it seems), then it's not a leak, but rather hard to collect cycles which ended up in 3rd generation of objects. Question is do we want to run full sweep more frequently or just ignore slightly higher memory usage?

**Comment by alexmbird** [comment]
I can't speak for everyone but my use case is deploying it to a Kubernetes cluster alongside the rest of our application stack, and I imagine that'll be a fairly common one.  Best practice is to give pods a fixed memory limit so while we aren't very resource-constrained we do need predictable usage.

As @kkopachev noted the usage for us has stabilised around 1GB per Thumbor instance.  That's a lot higher than before but we can live with it.  I had hoped the `GC_INTERVAL` setting would help us reduce it by triggering garbage collection differently.

Given `GC_INTERVAL` no longer does anything I think a simple solution would be to remove it from the documentation.  Lower memory usage is always welcome though!

**Comment by kkopachev** [comment]
@alexmbird Could you share your thumbor.conf? 
I've tried running a test server with request handlers which call `gc.collect` when I want to, and it always seem to report 0 objects collected. So I am thinking that your memory hike could be specific to a filter/optimizer or another module responsible for keeping some references around.

**Comment by alexmbird** [comment]
We're running the `minimalcompact/thumbor` Docker Hub image so the config file is generated automatically on startup.  I've extracted it from one of our pods:

[thumbor.conf.txt](https://github.com/thumbor/thumbor/files/3003167/thumbor.conf.txt)

**Comment by kkopachev** [comment]
Thanks @alexmbird !

Tried your config locally with debug handlers. 
Looks like "leak" is not related to storage and metrics; png and jpeg originals are fine also. However gif creates tons of collectable garbage after each request. I'll investigate further as time permits. 

As a workaround, you might want to try using gifsicle engine. I found it overall less memory consuming and producing better compressed gifs. `USE_GIFSICLE_ENGINE = True` in the config.
